.github/workflows/test.yml

```diff
@@ -0,0 +1,31 @@
+name: test
+
+on:
+  push:
+    branches:
+      - "*"
+  pull_request:
+    branches: [main]
+
+jobs:
+  build:
+    name: Build
+    runs-on: ubuntu-latest
```

Comment:
Overall I like the changes a lot, moving these commands to a Makefile makes it look a lot cleaner...I see that you moved these steps out of a dedicated workflow container. Is that mainly for simplification?

Reply:
Nope, I just copied what was in liatrio/rode-collector-sonarqube and made a few changes.  I'll work on getting this to run in a container, I don't think I've done that before with gh actions.